### PR TITLE
Add configuration for tox, poetry and mypy

### PR DESCRIPTION
**Pull request by WizzyGeek**
Also changed the project structure.

Have a look and let me know if it looks good.

@mradigen 
@Meetesh-Saini

### CODE REVIEW

File: pyproject.toml
Comment (mradigen) [inline pyproject.toml:1]: Wouldn't we have to include `py.typed` in `pyproject.toml` if we're making pwncore a package?

**Review by mradigen — APPROVED** [review]
Everything else LGTM

File: pyproject.toml
Comment (WizzyGeek) [reply pyproject.toml:1]: Poetry includes everything in the package source tree, no additional includes needed 👍🏼

File: src/pwncore/routes/__init__.py
Comment (Meetesh-Saini) [inline src/pwncore/routes/__init__.py:1]: Change the import to absolute

**Review by Meetesh-Saini — CHANGES_REQUESTED** [review]

**Comment by WizzyGeek** [comment]
All done, GTG @Meetesh-Saini @mradigen

**Comment by WizzyGeek** [comment]
Tox workflow is broken will create issue for that

**Review by Meetesh-Saini — APPROVED** [review]
Looks good to me

**Review by nexxeln — APPROVED** [review]

**Pull request merged**